### PR TITLE
Verification task per split

### PR DESCRIPTION
**Pull request by keyboardDrummer**
Dependent Dafny PR: https://github.com/dafny-lang/dafny/pull/5031

### Changes
- Change the API `executionEngine.GetImplementationTasks` to `executionEngine.GetVerificationTasks`, which returns a task for each statically defined assertion batch. When using the split on every assert option, this will return a verification task for each assertion.

### Testing
- Added test SplitOnEveryAssertion

### CODE REVIEW

File: Source/VCGeneration/Split.cs
Comment (atomb) [inline Source/VCGeneration/Split.cs:873]: Much nicer. :)

File: Source/UnitTests/ExecutionEngineTests/ExecutionEngineTest.cs
Comment (atomb) [inline Source/UnitTests/ExecutionEngineTests/ExecutionEngineTest.cs:70]: Yeah, I've been wondering about the empty first split for a while now. It'd be nice to get rid of it.

File: Source/UnitTests/ExecutionEngineTests/ExecutionEngineTest.cs
Comment (atomb) [inline Source/UnitTests/ExecutionEngineTests/ExecutionEngineTest.cs:368]: Given that at least this test of caching was working before, it'd be unfortunate to merge something that breaks it. Since I know that's something you're working on, do you have an upcoming PR already planned that'll fix it? It's perhaps worth making that separate for ease of reviewing, but I'd be reluctant to push a release of Boogie until it's at least working as well as it was before.

**Review by atomb — DISMISSED** [review]
I like the changes this makes a lot. I left a few comments that could result in changes, though not necessarily as part of this PR. The one critical one is that I'd only want to merge this with caching broken if you know you'll have another PR immediately following that fixes it (before the next Boogie release).

File: Source/UnitTests/ExecutionEngineTests/ExecutionEngineTest.cs
Comment (keyboardDrummer) [reply Source/UnitTests/ExecutionEngineTests/ExecutionEngineTest.cs:368]: Caching is currently not used by Dafny, because we don't trust it. This pathway of Boogie, for which caching does not work, is only used by Dafny. To 'fix' the caching, I would not to write some new code that does caching at the split level instead of the implementation level. Probably nothing complex, but it does not make sense to me to develop something that we won't test end-to-test because we're not using it.

File: Source/VCGeneration/VerificationRunResult.cs
Comment (keyboardDrummer) [reply Source/VCGeneration/VerificationRunResult.cs:11]: Fixed

File: Source/UnitTests/ExecutionEngineTests/ExecutionEngineTest.cs
Comment (atomb) [reply Source/UnitTests/ExecutionEngineTests/ExecutionEngineTest.cs:374]: If you're confident that it's not used by any other tools, then this sounds fine.

**Review by atomb — APPROVED** [review]

**Pull request merged**
